doc/src/build/fullnode.md

```diff
@@ -288,3 +288,119 @@ If you followed the instructions for [Building from Source](#building-from-sourc
     ```
 
 Your Full node starts on: `http://127.0.0.1:9000`.
+
+# Pruning 
+
+## Object Pruning
+
+Object pruning involves the removal of older object versions from the fullnode db. 
+New object versions are added to the db during transaction execution (making older versions ready for garbage collection) and can become a 
+problem if not pruned, as it can slow down the performance of the db and take up too much disk space. 
+Object pruning works in the background by tailing checkpoint stream, identifying each
+checkpoint which contains object versions which are now eligible for pruning.
+
+A sui node can enable object pruning by adding this `authority-store-pruning-config` config:
+```yaml
+authority-store-pruning-config:
+  # Number of epoch dbs to keep 
+  # Not relevant for object pruning
+  num-latest-epoch-dbs-to-retain: 3
+  # How often the epoch db pruning task should run
+  # Not relevant for object pruning
+  epoch-db-pruning-period-secs: 3600
+  # Number of latest epochs before which object pruning is allowed to run.
+  # When this is 0, pruner is allowed to prune old object versions as soon
+  # as possible. This is what we also call aggressive pruning and is the most effective
+  # garbage collection method which results in the samllest disk usage possible. 
+  # This is recommended setting for validator nodes as older object versions aren't
+  # needed for transaction execution.
+  # When this is 1, pruner will only prune object versions from transaction checkpoints
+  # older than the current epoch. In general, when this is N (where N >= 1), pruner 
+  # will only prune object versions from checkpoints upto current - N epoch. 
+  # This means it is possible to have multiple versions of an object to be present 
+  # in the db. It is a recommended setting for fullnodes as they may have to serve 
+  # rpc requests which require looking up objects by id and version (instead of just latest
+  # version). However, if a fullnode is not serving such rpc requests it should also 
+  # use aggressive pruning
```

Comment:
```suggestion
  # When set to1, Sui prunes only object versions from transaction checkpoints
  # previous to the current epoch. In general, when set to N (where N >= 1), Sui prunes  
  # only object versions from checkpoints up to `current - N` epoch. 
  # It is therefore possible to have multiple versions of an object present 
  # in the database. This setting is recommended for Sui Full nodes as they might need to serve 
  # RPC requests that require looking up objects by ID and Version (rather than just latest
  # version). However, if your Full node does not serve RPC requests you should then also enable  
  # aggressive pruning.
```